### PR TITLE
Reprogram aniso from scratch

### PR DESCRIPTION
**Pull request by tduretz**
Hi guys,

Anisotropy is getting a new treatment. Anisotropy related functions are in `AnisotropyRoutines.c`.
The default stress evaluation routine `NonNewtonianViscosityGrid` (`RheologyDensity.c`) does not contain anisotropy-related stuff. Default behaviour is thus isotropic (`aniso==0`).
If `aniso==1`, the engine now relies on `NonNewtonianViscosityGrid` to perform stress updates using an anisotropic model.
So far only an anisotropic viscous with constant viscosity is implemented (as in Williams's test). It is now implemented using the approach we designed with @halterw a few weeks ago.
@halterw: It would be great to check that we still have consistent results.
`ShearTemplate.txt` was updated accordingly.

If everything is fine, I will continue with anisotropic  power-law viscous, then visco-elastic and plastic implementations.

cheers!

### CODE REVIEW

**Comment by halterw** [comment]
Thanks, @tduretz! I'll check if we get the same results, then approve the pull-request! :)
Cheers, William

File: AnistropyUnitTests/data4plot.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+angle Txx Tzz Txz Tii
```
Comment (kulakovri) [inline AnistropyUnitTests/data4plot.txt]: Are those data files references?

Comment (kulakovri) [reply AnistropyUnitTests/data4plot.txt]: If not, we can store it in an extention that we can gitignore

File: AnistropyUnitTests/test_anisotropy_v0.c [hunk 2]
```diff
@@ -0,0 +1,331 @@
+#ifndef _GNU_SOURCE
```
Comment (kulakovri) [inline AnistropyUnitTests/test_anisotropy_v0.c]: Is it still part of the Pull Request?

File: MDLIB/AnisotropyRoutines.c [hunk 3]
```diff
@@ -0,0 +1,601 @@
...
+
+    //    for ( l=0; l<Ncz; l++ ) {
+    //        for ( k=0; k<Ncx; k++ ) {
+
+    // printf("average = %d\n", average);
+    k      = mesh->kp[k1];
+    l      = mesh->lp[k1];
+    c0     = k  + l*(Ncx);
+
+    mesh->eta_n[c0]       = 0.0;
```
Comment (kulakovri) [inline MDLIB/AnisotropyRoutines.c:384]: It did not get initiated to zero for each byte by DoodzCalloc?

File: MDLIB/AnisotropyRoutines.c [hunk 4]
```diff
@@ -0,0 +1,601 @@
...
+      
+      // Loop on phases
+      for ( p=0; p<model->Nb_phases; p++) {
+
+        // Detect if there is a fraction of phase p in the cell c: compute only if there is a non-zero fraction
+        bool is_phase_active = false;
+        const double min_fraction=1e-13;
+        if ( fabs(mesh->phase_perc_n[p][c0])>min_fraction ) is_phase_active = true;
+
+        if ( is_phase_active==true ) {
```
Comment (kulakovri) [inline MDLIB/AnisotropyRoutines.c]: In C a statement `is_phase_active==true` has the same meaning as just `is_phase_active` statement. So == true is obsolete

File: AnistropyUnitTests/data4plot.txt (hunk 1 above)
Comment (tduretz) [reply AnistropyUnitTests/data4plot.txt]: yes, let's git ignore it.

File: AnistropyUnitTests/test_anisotropy_v0.c (hunk 2 above)
Comment (tduretz) [reply AnistropyUnitTests/test_anisotropy_v0.c]: no, I will delete

File: MDLIB/AnisotropyRoutines.c (hunk 3 above)
Comment (tduretz) [reply MDLIB/AnisotropyRoutines.c:384]: yes, but this is within an interpolation. It need to be re-initialised before accumulating in it (at every iteration of every step)

File: MDLIB/AnisotropyRoutines.c (hunk 4 above)
Comment (tduretz) [reply MDLIB/AnisotropyRoutines.c]: Right, will delete

**Pull request merged**